test/integration_test.py

```diff
@@ -326,14 +335,28 @@ def _prune_test_bundles(self, bundle_fqids: List[BundleFQID], max_bundles: int)
                 break
         return filtered_bundle_fqids
 
+    def _filter_superseded_bundle_versions(self, bundle_fqids: Set[BundleFQID]) -> Set[BundleFQID]:
+        filtered_fqids = {
```

Comment:
No need for grouping and sorting if you're producing a dict. The dict is a grouping data structure in itself. Just use a `defaultdict(list)` and successively append to the values. Then use max to get the latest version. And try to avoid having to reconstruct the BundleFQID tuples by making the values lists of full BundleFQIDs and not just versions.